source/02_algo/model_based_rl_eng.rst

```diff
@@ -0,0 +1,147 @@
+Model-Based Reinforcement Learning
+######################################
+
+
+Model-Based Reinforcement Learning (Model-Based RL) is an important branch of reinforcement learning. The agent learns an environmental dynamics model through information interacting with the environment, and then uses the model to generate data to optimize action strategies, or use the model for planning.
+Model-Based RL methods first let the agent learn the environment dynamics model from the data obtained by interacting with the environment, and then use the environment model to generate a large number of virtual samples, thereby reducing the number of interactions with the real environment. Therefore, the sample utilization efficiency can be naturally improved.
+
+.. image:: ./images/model-based_rl.png
+  :align: center
+  :scale: 55%
+
+The environment model can generally be abstracted mathematically into a state transition function and a reward function.
+After learning the environment model, ideally, the agent does not need to interact with the real environment, but only in the simulated environment, which maximizes the cumulative discount reward through the RL algorithm to obtain the optimal policy.
+
+
+Problem Definition and Research Motivation
+----------------------------------------------
+
+In general, the problems of Model-Based RL research can be divided into two categories: one is how to learn an accurate environment model, and the other is how to use the environment model for policy optimization.
+
+**How to build an accurate environment model?** 
+
+Model learning mainly emphasizes the process of building an environment model by the Model-Based RL algorithm. For example, 
+
+  - `World Model <https://worldmodels.github.io/>`_ [3]_ proposes an environment model based on unsupervised learning and uses this model to transfer tasks from simulation to reality.
+  - `I2A <https://arxiv.org/abs/1707.06203>`_ [4]_ proposes an imagination-augmented-based model structure, based on which the future trajectory is predicted, and the trajectory information is encoded to assist policy learning.
+
+  But Model-Based RL also has several problems in the model learning part, for example,
+
+  - There will be errors in the environment model, and with the iterative interaction between the algorithm and the environment model, the accumulated error will become larger and larger, making it difficult for the algorithm to converge to the optimal solution.
```

Comment:
- There will be errors in the _**dynamics**_ model, and with the iterative interaction between the agent and the _**dynamics**_ model, the **_error induced by the model will compound over time_**, making it difficult for the algorithm to converge to the optimal solution.